### PR TITLE
feat: KG to work with Project Access Tokens

### PR DESCRIPTION
**Pull request by jachro**
This PR introduces the concept of Project Access Tokens.

### CODE REVIEW

File: graph-commons/src/main/scala/io/renku/http/client/AccessToken.scala [hunk 1]
```diff
@@ -28,40 +28,60 @@ import java.util.Base64
...
+      extends TinyTypeFactory[ProjectAccessToken](new ProjectAccessToken(_))
+      with NonBlank[ProjectAccessToken]
+
   private val base64Decoder = Base64.getDecoder
   private val base64Encoder = Base64.getEncoder
 
   implicit val accessTokenEncoder: Encoder[AccessToken] = {
+    case ProjectAccessToken(token) =>
+      Json.obj("projectAccessToken" -> Json.fromString(new String(base64Encoder.encode(token.getBytes(UTF_8)), UTF_8)))
     case OAuthAccessToken(token) =>
```
Comment (eikek) [inline graph-commons/src/main/scala/io/renku/http/client/AccessToken.scala]: Now the name `OAuth…` seems a bit too generic. It is only for a user token used. If understood correctly the "project token" is an oauth token as well (being used in an oauth flow)? (nit picking obviously :))

**Review by eikek — APPROVED** [review]
👍🏼

File: graph-commons/src/main/scala/io/renku/http/client/AccessToken.scala (hunk 1 above)
Comment (jachro) [reply graph-commons/src/main/scala/io/renku/http/client/AccessToken.scala]: True. Perhaps, it'd be better to name it `UserOAuthAccessToken`? What do you think?

Comment (eikek) [reply graph-commons/src/main/scala/io/renku/http/client/AccessToken.scala]: Yes, that sounds good 👍🏼

Comment (jachro) [reply graph-commons/src/main/scala/io/renku/http/client/AccessToken.scala]: 👍🏻

**Pull request merged**